irispy/new_sji.py

```diff
@@ -136,6 +137,78 @@ def __repr__(self):
                axis_types=self.world_axis_physical_types,
                dimensions=self.dimensions))
 
+    def apply_exposure_time_correction(self, undo=False, copy=False, force=False):
+        """
+        Applies or undoes exposure time correction to data and uncertainty and adjusts unit.
+
+        Correction is only applied (undone) if the object's unit doesn't (does)
+        already include inverse time.  This can be overridden so that correction
+        is applied (undone) regardless of unit by setting force=True.
+
+        Parameters
+        ----------
+        undo: `bool`
+            If False, exposure time correction is applied.
+            If True, exposure time correction is removed.
+            Default=False
+
+        copy: `bool`
+            If True a new instance with the converted data values is returned.
+            If False, the current instance is overwritten.
+            Default=False
+
+        force: `bool`
+            If not True, applies (undoes) exposure time correction only if unit
+            doesn't (does) already include inverse time.
+            If True, correction is applied (undone) regardless of unit.  Unit is still
+            adjusted accordingly.
+
+        Returns
+        -------
+        result: `None` or `SJICube`
+            If copy=False, the original SJICube is modified with the exposure
+            time correction applied (undone).
+            If copy=True, a new SJICube is returned with the correction
+            applied (undone).
+
+        """
+        # Get exposure time in seconds and change array's shape so that
+        # it can be broadcast with data and uncertainty arrays.
+        exposure_time_s = u.Quantity(self.extra_coords["EXPOSURE TIME"]["value"], unit='s').value
+        if not isinstance(self.extra_coords["EXPOSURE TIME"]["value"], float):
```

Comment:
Instead of ```isinstance(..., float)```,  this should be ```np.isscalar(...)```.  This way we don't get caught by edges cases where the value is ```int```, ```np.int64```, etc.

Reply:
Modified